### DISCUSSION TITLE
Custom logic and form for CREATE action in Filament Admin Panel

### DISCUSSION THREAD

**Discussion by AnoNimo722** [filamentphp/filament#14908] [Help] [answered]
### Package

Panel builder

### Package Version

3.2.121

### How can we help you?

Good day everyone!

I would like to know if anything has changed in the handling of filament forms in the livewire component? With Laravel 10 the operation worked correctly, but now in Laravel v11 it doesn't work for me, I keep getting an HTTP 500 error. Maybe it's my fault and I missed something.

I want to change form's inputs and the default logic of handling the create action for the Account model. If I use the form fields in the html form in the blade view, there is no problem. The problem is with the `{{ $this->form }}` and or with `return $form->schema([
                View('livewire.create-account-form'),
            ]);` method itself, nothing is written in the Laravel log and I have no more ideas what could have gone wrong.

Or maybe I'm going in the wrong direction and should do it differently?

I would be very grateful for your advice, warm regards

AccountResource:
```
class AccountResource extends Resource
{
    protected static ?string $model = Account::class;
    
    public static function form(Forms\Form $form): Forms\Form
    {
         $isCreate = request()->routeIs('filament.admin.resources.accounts.create');
          if ($isCreate) {
            return $form->schema([
                Forms\Components\View::make('livewire.create-account-form'),
            ]);
            }
    }
```

Livewire CreateAccountForm:
```
<?php

namespace App\Livewire;

use Livewire\Component;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Forms\Form;
use Illuminate\Contracts\View\View;
use Filament\Forms\Components\TextInput;
use Illuminate\Support\Facades\Log;

class CreateAccountForm extends Component implements HasForms
{

    use InteractsWithForms;

    public $name;
    public ?array $data = [];

    public function mount(): void
    {
        $this->form->fill();
    }

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name')
                    ->required(),
            ])
            ->statePath('data');
    }

    public function create(): void
    {
        Log::Info("create");
    }

    public function render(): View
    {
        return view('livewire.create-account-form');
    }
}
```

create-account-form.blade.php
```
<div>
    <form wire:submit="create">
        {{ $this->form }}

        <button type="submit">
            Submit
        </button>
    </form>

    <x-filament-actions::modals />
</div>
```

I am using default routes:
```
public static function getPages(): array
    {
        return [
            'index' => Pages\ListAccounts::route('/'),
            'create' => Pages\CreateAccount::route('/create'),
            'edit' => Pages\EditAccount::route('/{record}/edit'),
        ];
    }
```

**Comment by leandrocfe** [comment]
You don't need to render a custom form, just use `operation` to show/hide fields according to the Page

```php
public static function form(Form $form): Form
{
    return $form->schema([
        Group::make([
            TextInput::make('name')
                ->label('Name on Create'),
        ])
            ->visibleOn('create'),
        Group::make([
            TextInput::make('name')
                ->label('Name on Edit'),
        ])
            ->visibleOn('edit'),
    ]);
}
```

**Reply by AnoNimo722** [reply]
thanks for the tip, apart from different form fields I also need different creation logic, because I don't store all the data in one table/database, and the connections in the model from what I see are not enough and records that should be stored in other tables are not created.

for example:
```
public function emails()
{
return $this->belongsToMany(Email::class, 'email_ids', 'a_id', 'email_id');
}
```
belongsToMany relationship, because I need to store the last few addresses assigned to the account

**Reply by leandrocfe** [reply, marked as answer]
You can also [customize the creation process](https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-the-creation-process)

**Reply by AnoNimo722** [reply]
I just came across this in the documentation, I can't believe I missed it before

Thank you very much :)
